StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/IntentConfirmParamsTest.swift

```diff
@@ -15,23 +15,23 @@ class IntentConfirmParamsTest: XCTestCase {
         let intentConfirmParams = IntentConfirmParams(type: .stripe(.card))
 
         intentConfirmParams.saveForFutureUseCheckboxState = .hidden
-        intentConfirmParams.setAllowRedisplay(paymentMethodSave: nil, isSettingUp: true)
+        intentConfirmParams.setAllowRedisplay(paymentMethodSave: nil, allowRedisplayOverride: nil, isSettingUp: true)
 
         XCTAssertEqual(.unspecified, intentConfirmParams.paymentMethodParams.allowRedisplay)
     }
     func testSetAllowRedisplay_legacyPI_selected() {
         let intentConfirmParams = IntentConfirmParams(type: .stripe(.card))
 
         intentConfirmParams.saveForFutureUseCheckboxState = .selected
-        intentConfirmParams.setAllowRedisplay(paymentMethodSave: nil, isSettingUp: false)
+        intentConfirmParams.setAllowRedisplay(paymentMethodSave: nil, allowRedisplayOverride: nil, isSettingUp: false)
 
         XCTAssertEqual(.unspecified, intentConfirmParams.paymentMethodParams.allowRedisplay)
     }
     func testSetAllowRedisplay_legacyPI_deselected() {
         let intentConfirmParams = IntentConfirmParams(type: .stripe(.card))
 
         intentConfirmParams.saveForFutureUseCheckboxState = .deselected
-        intentConfirmParams.setAllowRedisplay(paymentMethodSave: nil, isSettingUp: false)
+        intentConfirmParams.setAllowRedisplay(paymentMethodSave: nil, allowRedisplayOverride: nil, isSettingUp: false)
```

Comment:
Very much a rough thought but: these tests deeply bug me but I'm not exactly sure how to explain why and I don't have the knowledge to suggest something better. Like, we're just testing every possible combination of parameters right?

Reply:
Maybe we do bite the bullet of testing all input combinations, in which case perhaps we should at least do like (copied from https://stackoverflow.com/questions/35441124/parametrized-unit-tests-in-swift):

```
func testDivision() {
    let testCases = [
        (a: 12, b: 3, result: 4),
        (a: 12, b: 2, result: 6),
        (a: 12, b: 6, result: 1),
        (a: 12, b: 4, result: 3),
    ]
    for (a, b, result) in testCases {
        XCTContext.runActivity(named: "Testing dividing \(a) by \(b) to get result \(result)") { activity in
            XCTAssertEqual(result, a/b)
        }
    }
}
```

Reply:
We are testing all use cases that we support (and two cases that technically shouldn't happen, but we are testing that our code is defensive enough to handle it).

Separating out each test into a separate function allows for narrowing down what exact use case is broken.

Reply:
Yeah I guess my suggestion is that the example above is IMO easier to read than separate testcases, and the `XCTContext.runActivity(named:` line lets you know the exact test case that breaks.